blocks/link-editor/index.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * External dependencies
+ */
+import { isBoolean, noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+import { IconButton } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import UrlInput from '../url-input';
+import { filterURLForDisplay } from '../../editor/utils/url';
+import ToggleControl from '../inspector-controls/toggle-control';
+
+export default class LinkEditor extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			settingsVisible: false,
+			newLinkValue: '',
+			isEditing: null,
+		};
+
+		this.onChangeLinkValue = this.onChangeLinkValue.bind( this );
+		this.toggleLinkSettingsVisibility = this.toggleLinkSettingsVisibility.bind( this );
+		this.submitLink = this.submitLink.bind( this );
+		this.editClicked = this.editClicked.bind( this );
+		this.dropClicked = this.dropClicked.bind( this );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( isBoolean( nextProps.isEditing ) ) {
```

Comment:
I think you're not using this prop in this PR. Is this something you think we'll use later?

Reply:
Yes, for example when using the component in the formatting toolbar, it is useful to control the edition state from the toolbar. With a prop, we can do that.